PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ['digest.digest']
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']
-    _inherit = ['sale.order']
-    _inherit = ["sale.order.line"]

Total Changes: 14

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_management/models/digest.py
+++ b/addons/sale_management/models/digest.py
@@ -5,7 +5,7 @@ from odoo.exceptions import AccessError
 
 
 class DigestDigest(models.Model):
-    _inherit = ['digest.digest']
+    _inherit = 'digest.digest'
 
     kpi_all_sale_total = fields.Boolean('All Sales')
     kpi_all_sale_total_value = fields.Monetary(compute='_compute_kpi_sale_total_value')

--- a/addons/sale_management/models/res_company.py
+++ b/addons/sale_management/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
     _check_company_auto = True
 
     sale_order_template_id = fields.Many2one(

--- a/addons/sale_management/models/res_config_settings.py
+++ b/addons/sale_management/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     group_sale_order_template = fields.Boolean(
         "Quotation Templates", implied_group='sale_management.group_sale_order_template')

--- a/addons/sale_management/models/sale_order.py
+++ b/addons/sale_management/models/sale_order.py
@@ -8,7 +8,7 @@ from odoo.tools import is_html_empty
 
 
 class SaleOrder(models.Model):
-    _inherit = ['sale.order']
+    _inherit = 'sale.order'
 
     sale_order_template_id = fields.Many2one(
         comodel_name='sale.order.template',

--- a/addons/sale_management/models/sale_order_line.py
+++ b/addons/sale_management/models/sale_order_line.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class SaleOrderLine(models.Model):
-    _inherit = ["sale.order.line"]
+    _inherit = "sale.order.line"
     _description = "Sales Order Line"
 
     sale_order_option_ids = fields.One2many('sale.order.option', 'line_id', 'Optional Products Lines')

--- a/addons/sale_management/models/sale_order_option.py
+++ b/addons/sale_management/models/sale_order_option.py
@@ -5,6 +5,7 @@ from odoo.exceptions import UserError
 
 
 class SaleOrderOption(models.Model):
+    _name = 'sale.order.option'
     _description = "Sale Options"
     _order = 'sequence, id'
 

--- a/addons/sale_management/models/sale_order_template.py
+++ b/addons/sale_management/models/sale_order_template.py
@@ -6,6 +6,7 @@ from odoo.fields import Command
 
 
 class SaleOrderTemplate(models.Model):
+    _name = 'sale.order.template'
     _description = "Quotation Template"
     _order = 'sequence, id'
 

--- a/addons/sale_management/models/sale_order_template_line.py
+++ b/addons/sale_management/models/sale_order_template_line.py
@@ -5,6 +5,7 @@ from odoo.exceptions import UserError
 
 
 class SaleOrderTemplateLine(models.Model):
+    _name = 'sale.order.template.line'
     _description = "Quotation Template Line"
     _order = 'sale_order_template_id, sequence, id'
 

--- a/addons/sale_management/models/sale_order_template_option.py
+++ b/addons/sale_management/models/sale_order_template_option.py
@@ -4,6 +4,7 @@ from odoo import api, fields, models
 
 
 class SaleOrderTemplateOption(models.Model):
+    _name = 'sale.order.template.option'
     _description = "Quotation Template Option"
     _check_company_auto = True
 
